net/dns/manager.go

```diff
@@ -454,17 +463,61 @@ func (s *dnsTCPSession) handleReads() {
 // servicing DNS requests sent down it.
 func (m *Manager) HandleTCPConn(conn net.Conn, srcAddr netaddr.IPPort) {
 	s := dnsTCPSession{
-		m:            m,
-		conn:         conn,
-		srcAddr:      srcAddr,
-		responses:    make(chan []byte),
-		readClosing:  make(chan struct{}),
+		m:           m,
+		conn:        conn,
+		srcAddr:     srcAddr,
+		responses:   make(chan []byte),
+		readClosing: make(chan struct{}),
 	}
 	s.ctx, s.closeCtx = context.WithCancel(context.Background())
 	go s.handleReads()
 	s.handleWrites()
 }
 
+// HandleTCPConn implements magicDNS over DNS-over-TLS, taking a connection and
+// servicing DNS requests sent down it.
+func (m *Manager) HandleDNSoverTLSConn(conn net.Conn, srcAddr netaddr.IPPort) {
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
This will work in Android's PrivateDNS=Automatic mode, where it opportunistically tries to use DoT and does not validate the certificate it receives.

With Android's PrivateDNS=On setting I don't think this will be sufficient, it wants to see a real certificate signed by a root of trust. We should be able to do that eventually if the tailnet has certificate support set up and we're willing to proactively obtain a cert for android-name.tails-scales.ts.net.